aiohttp/client_exceptions.py

```diff
@@ -238,17 +239,34 @@ class InvalidURL(ClientError, ValueError):
 
     # Derive from ValueError for backward compatibility
 
-    def __init__(self, url: Any) -> None:
+    def __init__(self, url: StrOrURL, description: Optional[str] = None) -> None:
         # The type of url is not yarl.URL because the exception can be raised
         # on URL(url) call
-        super().__init__(url)
+        if description:
+            super().__init__(url, description)
+        else:
+            super().__init__(url)
+
+    @property
+    def url(self) -> StrOrURL:
+        return self.args[0]  # type: ignore[no-any-return]
```

Comment:
```suggestion
        return self.args[0]
```

Reply:
If the ignore wasn't needed, it'd error in mypy. In this case args are from the base exception and are therefore `Any`. Maybe... we can just define the type in the subclass with `args: tuple[StrOrUrl, str]` or similar.

Reply:
Can we resolve this conversation?

Reply:
Okay, marking it as resolved.